src/Button/DigitizeButton/DigitizeButton.jsx

```diff
@@ -1018,22 +1022,37 @@ class DigitizeButton extends React.Component {
         this._digitizeFeatures.remove(this._digitizeTextFeature);
         this._digitizeTextFeature = null;
       }
-      if (onModalLabelCancel) {
+      if (isFunction(onModalLabelCancel)) {
         onModalLabelCancel();
       }
     });
   }
 
   /**
    * Sets formatted label on feature.
+   * Calls `onModalLabelOk` callback function if provided.
    *
    * @param {OlFeature} feat The point feature to be styled with label.
+   * @param {Function} onModalOkCbk Optional callback function.
    */
-  setTextOnFeature = feat => {
-    const label = StringUtil.stringDivider(this.state.textLabel, 16, '\n');
+  setTextOnFeature = (feat, onModalOkCbk) => {
+    const {
+      maxLabelLineLength
+    } = this.props;
+
+    let label = this.state.textLabel;
+    if (maxLabelLineLength) {
+      label = StringUtil.stringDivider(
+        this.state.textLabel, maxLabelLineLength, '\n'
```

Comment:
Haven't checked, what `stringDivider` does in detail, but I would suggest to pass a cloned or destructured `textLabel` string to the method to ensure, the state object won't be manipulated in it.